Sources/ContentType.swift

```diff
@@ -178,4 +178,28 @@ extension ContentType: ResourceQueryable {
                             }
         })
     }
+
+    public func fetch<ResourceType>() async throws -> (Result<ResourceType, Error>, ResponseType)
+    where ResourceType: EndpointAccessible, ResourceType: Decodable {
+        guard let uid = self.uid else { fatalError("Please provide ContentType uid") }
+        
+        do {
+            let (data, response): (Result<ContentstackResponse<ResourceType>, Error>, ResponseType) = try await self.stack.asyncFetch(endpoint: ResourceType.endpoint,
+                                                                   cachePolicy: self.cachePolicy,
+                                                                   parameters: parameters + [QueryParameter.uid: uid],
+                                                                   headers: headers)
+            switch data {
+            case .success(let contentstackResponse):
+                if let resource = contentstackResponse.items.first {
+                    return (.success(resource), response)
+                } else {
+                    return (.failure(SDKError.invalidUID(string: uid)), response)
+                }
+            case .failure(let error):
+                return (.failure(error), response)
```

Comment:
Failure should throw error here instead of error @vkalta 

**Note** : apply it for all